internal/weldr/api.go

```diff
@@ -2025,6 +2026,67 @@ func (api *API) composeImageHandler(writer http.ResponseWriter, request *http.Re
 	common.PanicOnError(err)
 }
 
+// composeMetadataHandler returns a tar of the metadata used to compose the requested UUID
+func (api *API) composeMetadataHandler(writer http.ResponseWriter, request *http.Request, params httprouter.Params) {
+	if !verifyRequestVersion(writer, params, 0) {
+		return
+	}
+
+	uuidString := params.ByName("uuid")
+	uuid, err := uuid.Parse(uuidString)
+	if err != nil {
+		errors := responseError{
+			ID:  "UnknownUUID",
+			Msg: fmt.Sprintf("%s is not a valid build uuid", uuidString),
+		}
+		statusResponseError(writer, http.StatusBadRequest, errors)
+		return
+	}
+
+	compose, exists := api.store.GetCompose(uuid)
+	if !exists {
+		errors := responseError{
+			ID:  "UnknownUUID",
+			Msg: fmt.Sprintf("Compose %s doesn't exist", uuidString),
+		}
+		statusResponseError(writer, http.StatusBadRequest, errors)
+		return
+	}
+
+	composeStatus := api.getComposeStatus(compose)
+	if composeStatus.State != common.CFinished && composeStatus.State != common.CFailed {
+		errors := responseError{
+			ID:  "BuildInWrongState",
+			Msg: fmt.Sprintf("Build %s is in wrong state: %s", uuidString, composeStatus.State.ToString()),
+		}
+		statusResponseError(writer, http.StatusBadRequest, errors)
+		return
+	}
+
+	metadata, err := json.Marshal(compose)
```

Comment:
I think we should just expose the manifest here (the `Compose` is really internal state). So I would use `compose.ImageBuild.Manifest` here instead.

Reply:
Ok, added that, but also a check for nil that returns a 400 with an 'EmptyManifest' error.